### DISCUSSION TITLE
OpenRadioss implicit dynamic simulation

### DISCUSSION THREAD

**Discussion by kv-msuter** [OpenRadioss/OpenRadioss#1368] [Q&A]
Has anyone had success using OpenRadioss's dynamic implicit solver options for modeling real timescale events with reasonably large timesteps (~0.01sec)? I have been able to simulate a plastic bottle topload using the general Newmark solver (/IMPL/DYNA/2, gamma=0.6, beta=0.38) but the timestep is very low, in the realm of an explicit timestep (~2e-6 seconds). The simulation event time is 6 seconds but the solution diverges at about 2 seconds.

**Comment by Nekativ** [comment]
I have successfully used the implicit solver to preload a simulation before running an explicit simulation. The pre-simulation included several contacts and large deformations so it was not a simple case either, however it did require the usual head-knocking to get it to converge like any other implicit simulation.

Implicit sims can be very difficult to converge even with solvers that are dedicated to solving implicitly; So the first question is how difficult would this simulation be to converge with a full robust implicit solver? Without knowing the details of the simulation it is hard to give any specific answers.

**Buckling/Snap-Through**
The simulation as you describe is "plastic bottle topload". This sounds like it includes buckling and snap through. Snap through is when there is a sudden change in the geometry and shifting of load directions just as the bottle starts to buckle. 

If the load in reaches a point in your simulation reaches the buckling point of the bottle, then it is no mystery as to why it is diverging. A buckling simulation with snap through is very difficult to solve implicitly, the way to do it is to divide the timestep by a factor or two at the time in which the buckling occurs, I would argue this is better solved explicitly since the timesteps will be extremely small. 


**Top And Bottom Contacts**
If you have contact on the top and bottom of the bottle and it is changing heavily, in my experience this will be very upsetting to an implicit solver. You could eliminate these contacts and add them back in once you get everything else to converge. (Simplify you simulation and add back in complexity one step at a time.)

**Reply by kv-msuter** [reply]
Hi, Nekativ, thanks for your response and apologies for the delay in mine. We often have similar bouts of "head-knocking" to get our implicit solutions to converge in other solvers as well. Our bread-and-butter solver for these use cases have been dynamic implicit in Abaqus/Standard or LS-Dyna which for whatever reason generally reach a solution much quicker due to larger increments even given the inherent difficulties due to buckling and contact (or at least use large increments right up until the point of buckling), while OpenRadioss starts the solution already at that very small timestep. The long event time generally led us to stick with the implicit solver over an explicit one.

We've tried a large range of the troubleshooting methods you suggest and have not made a lot of progress with solution efficiency improvements with OpenRadioss dynamic implicit. My post was intended to see if there were specific solver settings others have had success with, but perhaps there is no silver bullet for this use case.

**Comment by andy4431** [comment]
Hi,
I would suggest running your model explicitly and applying the load "slowly" such that you minimize kinetic energy..  The smooth curve function, /FUNC_SMOOTH, can be useful for eliminating dynamic effects.  What is slow depends on the loading event.  First apply the load quick to check your general model setup and run time.  Then run it slower.  

Also, you can utilize mass scaling to add a small amount of mass to the model to increase the time step.  Again you sometimes have to do a check run with a few different mass amounts to make sure the added mass doesn't change your results.   You can specify the percent mass added using,
/DT/NODA/CST
0.9 0.0 0.02

Where the last number is the mass scaling amount so 0.02 is 2% mass added.
